test/chplcheck/CaseRules.chpl

```diff
@@ -0,0 +1,41 @@
+module CaseRules {
+    var camelVar: int;
+    const PascalVar: real;
+    var snake_var: string;
+
+    record camelRecord {
+        proc camelCaseMethod() {return 0;}
+        proc PascalCaseMethod() {return 0;}
+        proc snake_case_method() {return 0;}
+    }
```

Comment:
Note that last I checked the rule had a special exception for single-letter records (e.g. `R`). Whether or not this exception is kept, it might be an interesting case to test it here.